api/api.go

```diff
@@ -35,120 +30,115 @@ func HttpPOST(url string, data string) (resp *http.Response, err error) {
 	return new(http.Client).Do(req)
 }
 
-// handleHTTPRequest is a wrapper function that handles all incoming calls to
-// the API.
-func (srv *Server) handleHTTPRequest(mux *http.ServeMux, url string, handler http.HandlerFunc) {
-	mux.HandleFunc(url, func(w http.ResponseWriter, req *http.Request) {
-		if !strings.Contains(req.UserAgent(), srv.requiredUserAgent) {
+// requireUserAgent is middleware that requires all requests to set a
+// UserAgent that contains the specified string.
+func requireUserAgent(h http.Handler, ua string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if !strings.Contains(req.UserAgent(), ua) {
 			writeError(w, "Browser access disabled due to security vulnerability. Use Sia-UI or siac.", http.StatusBadRequest)
 			return
 		}
-		handler(w, req)
+		h.ServeHTTP(w, req)
 	})
 }
 
 // initAPI determines which functions handle each API call.
 func (srv *Server) initAPI() {
-	mux := http.NewServeMux()
-
-	// 404 Calls
-	srv.handleHTTPRequest(mux, "/", srv.unrecognizedCallHandler)
+	mux := httprouter.New()
+	mux.NotFound = http.HandlerFunc(srv.unrecognizedCallHandler) // custom 404
 
 	// Daemon API Calls - Unfinished
-	srv.handleHTTPRequest(mux, "/daemon/constants", srv.daemonConstantsHandler)
-	srv.handleHTTPRequest(mux, "/daemon/version", srv.daemonVersionHandler)
-	srv.handleHTTPRequest(mux, "/daemon/stop", srv.daemonStopHandler)
-	srv.handleHTTPRequest(mux, "/daemon/updates/apply", srv.daemonUpdatesApplyHandler)
-	srv.handleHTTPRequest(mux, "/daemon/updates/check", srv.daemonUpdatesCheckHandler)
+	mux.GET("/daemon/constants", srv.daemonConstantsHandler)
+	mux.GET("/daemon/version", srv.daemonVersionHandler)
+	mux.POST("/daemon/stop", srv.daemonStopHandler)
 
 	// Consensus API Calls
 	if srv.cs != nil {
-		srv.handleHTTPRequest(mux, "/consensus", srv.consensusHandler)            // GET
-		srv.handleHTTPRequest(mux, "/consensus/block", srv.consensusBlockHandler) // GET
+		mux.GET("/consensus", srv.consensusHandler)
+		mux.GET("/consensus/blocks/:height", srv.consensusBlocksHandler)
```

Comment:
I suppose that's fair, but they're not true resources in the first place because you can't PUT or DELETE them. I just like the idea that each conceptual "thing" (block, peer, address, etc.) is represented by a path, rather than a query string.
How about `/consensus/blockchain/<height>` (or just `chain`) for accessing blocks by height, and `/consensus/block/<id>` for accessing by id?